src/test_runner.cpp

```diff
@@ -0,0 +1,118 @@
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <iostream>
+
+#include "ast/element.h"
+#include "ast/span.h"
+#include "evaluator/evaluator.h"
+#include "evaluator/program.h"
+#include "reader/error.h"
+#include "reader/parser.h"
+#include "reader/reader.h"
+#include "reader/scanner.h"
+#include "reader/token.h"
+
+using reader::Reader;
+
+bool path_ends_with(std::filesystem::path path, std::string end) {
+    if (path.string().length() < end.length()) {
+        return false;
+    }
+
+    return path.string().substr(path.string().length() - end.length()) == end;
+}
+
+bool test_fail_file(std::filesystem::path path) {
+    std::ifstream file(path);
+    std::string line;
+
+    bool passed = true;
+
+    while (std::getline(file, line)) {
+        try {
+            Reader reader((std::string_view(line)));
+            auto elements = reader.read();
+            return false;
+        } catch (reader::SyntaxError e) {
+        }
+    }
+
+    file.close();
+
+    return passed;
+}
+
+std::vector<std::string> split(std::string str, char separator) {
+    std::vector<std::string> strings;
+
+    int startIndex = 0;
+    int endIndex = 0;
+    for (int i = 0; i <= str.size(); i++) {
+        if (str[i] == separator || i == str.size()) {
+            endIndex = i;
+            std::string temp = str.substr(startIndex, endIndex - startIndex);
+            startIndex = endIndex + 1;
+        }
+    }
+    return strings;
+}
+
+bool test_correct_file(std::filesystem::path path) {
+    std::ifstream file(path);
+    std::stringstream buffer;
+    buffer << file.rdbuf();
+    std::string source(buffer.str());
+
+    bool passed = true;
+
+    Reader reader((std::string_view(source)));
```

Comment:
стоит либо переписать без passed, либо хотя бы сделать очевиднее, что значение `passed` определяется в try/catch

```suggestion
    Reader reader((std::string_view(source)));

    bool passed = true;
```